.github/workflows/tests.yaml

```diff
@@ -0,0 +1,75 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - "ubuntu-*"
+  pull_request:
+    branches:
+      - "ubuntu-*"
+  schedule:
+    # run at 00:00 on the first of every month
+    - cron: "0 0 1 * *"
+
+jobs:
+  planner:
+    name: Determine jobs
+    runs-on: ubuntu-latest
+    outputs:
+      # install-all is set to 'true' if chisel.yaml changes
+      install-all: ${{ steps.changed-paths.outputs.chisel-yaml }}
+      # install-changed is set to 'true' if any slice definition files
+      # have been added or modified (not deleted).
+      install-changed: ${{ steps.changed-paths.outputs.slices }}
+      # changed-sdf is a space separated list of slice definition file
+      # paths who have been added or modified (not deleted).
+      changed-sdf: ${{ steps.changed-paths.outputs.slices_files }}
+    steps:
+      - uses: actions/checkout@v4
+        if: ${{ github.event_name != 'schedule' }}
+        with:
+          fetch-depth: 0
+
+      - name: Check changed paths
```

Comment:
the rationale is good, but it is slightly more complicated, but can use less lines of code. Eg.:

```yaml
        id: changed-paths
        if: ${{ github.event_name != 'schedule' }}
        uses: dorny/paths-filter@v3
        with:
          # ref: https://github.com/marketplace/actions/paths-changes-filter
          filters: |
            install-all:
              - 'chisel.yaml'
              - '.github/**'
              - deleted: 'slices/**'
            slices:
              - added|modified: 'slices/**/*.yaml'
          # Space delimited list usable as command-line argument list in
          # Linux shell. If needed, it uses single or double quotes to
          # wrap filename with unsafe characters.
          list-files: shell
```

and you can drop the `install-changed` output

Reply:
Updated, thanks!